<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" /><meta name="author" content="Shaojun Xie @ CCRSFIFX" />
      <link rel="shortcut icon" href="../img/favicon.ico" />
    <title>Demultiplexing and Analyzing 5’ Immune Profiling Libraries Pooled with Hashtags - scMaestro Documentation</title>
    <link rel="stylesheet" href="../css/theme.css" />
    <link rel="stylesheet" href="../css/theme_extra.css" />
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/styles/github.min.css" />
    
      <script>
        // Current page data
        var mkdocs_page_name = "Demultiplexing and Analyzing 5\u2019 Immune Profiling Libraries Pooled with Hashtags";
        var mkdocs_page_input_path = "multi_5p.md";
        var mkdocs_page_url = null;
      </script>
    
    <!--[if lt IE 9]>
      <script src="../js/html5shiv.min.js"></script>
    <![endif]-->
      <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/highlight.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/yaml.min.js"></script>
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.8.0/languages/rust.min.js"></script>
      <script>hljs.highlightAll();</script> 
</head>

<body class="wy-body-for-nav" role="document">

  <div class="wy-grid-for-nav">
    <nav data-toggle="wy-nav-shift" class="wy-nav-side stickynav">
    <div class="wy-side-scroll">
      <div class="wy-side-nav-search">
          <a href=".." class="icon icon-home"> scMaestro Documentation
        </a><div role="search">
  <form id ="rtd-search-form" class="wy-form" action="../search.html" method="get">
      <input type="text" name="q" placeholder="Search docs" aria-label="Search docs" title="Type search term here" />
  </form>
</div>
      </div>

      <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="Navigation menu">
              <p class="caption"><span class="caption-text">Introduction</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="..">About</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../design/">Design</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../custom_setting/">Custom setting</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Usage</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../gex/">Subcommand `rna`</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../multi/">Subcommand `multi`</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../vdj/">Subcommand `vdj`</a>
                  </li>
                  <li class="toctree-l1"><a class="reference internal" href="../fixedrna/">Subcommand `fixedrna`</a>
                  </li>
              </ul>
              <p class="caption"><span class="caption-text">Todo</span></p>
              <ul>
                  <li class="toctree-l1"><a class="reference internal" href="../todo/">todo</a>
                  </li>
              </ul>
              <ul>
                <li class="toctree-l1"><a class="reference internal" href="../license/">License</a>
                </li>
              </ul>
      </div>
    </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">
      <nav class="wy-nav-top" role="navigation" aria-label="Mobile navigation menu">
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="..">scMaestro Documentation</a>
        
      </nav>
      <div class="wy-nav-content">
        <div class="rst-content"><div role="navigation" aria-label="breadcrumbs navigation">
  <ul class="wy-breadcrumbs">
    <li><a href=".." class="icon icon-home" aria-label="Docs"></a></li>
      <li class="breadcrumb-item active">Demultiplexing and Analyzing 5’ Immune Profiling Libraries Pooled with Hashtags</li>
    <li class="wy-breadcrumbs-aside">
          <a href="https://github.com/CCRSF-IFX/scMaestro/edit/main/docs/multi_5p.md">Edit on CCRSF-IFX/scMaestro</a>
    </li>
  </ul>
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
            <div class="section" itemprop="articleBody">
              
                <h1 id="demultiplexing-and-analyzing-5-immune-profiling-libraries-pooled-with-hashtags">Demultiplexing and Analyzing 5’ Immune Profiling Libraries Pooled with Hashtags</h1>
<h2 id="step1-demultiplexing">Step1: demultiplexing</h2>
<h3 id="prepare-configpy-file">Prepare <code>config.py</code> file</h3>
<pre><code>unaligned=[&quot;/scratch/ccrsf_scratch/scratch/Illumina_Demultiplex/NovaSeq/20240816_LH00584_0065_B22NK2FLT3/22NK2FLT3/outs/fastq_path/22NK2FLT3&quot;]
analysis=&quot;/mnt/ccrsf-static/singlecell_projects/PamelaSchwartzberg_CS037516_9sclibs_08072024/Analysis_demultiplex&quot;
ref=&quot;mm10&quot;
projectname=&quot;PamelaSchwartzberg_CS037516_9sclibs_08072024&quot;
yields=&quot;1411633.0&quot;
archive=True
runs=&quot;20240816_LH00584_0065_B22NK2FLT3&quot;
libraries=&quot;libraries.csv&quot;
pipeline=&quot;multi&quot;
cmo=&quot;cmo.csv&quot;
aggregate=False
</code></pre>
<h3 id="prepare-librariescsv">Prepare <code>libraries.csv</code></h3>
<p>In <code>libraries.csv</code>, only <code>Gene Expression</code> and <code>Multiple Capture</code> samples are included to run <code>multi</code> pipeline. The purpose of this step is demultiplexing. </p>
<pre><code>Name,Flowcell,Sample,Type
CD8_d8_1,22NK2FLT3,1_cDNA_CD8_d8_1,Gene Expression
CD8_d8_2,22NK2FLT3,2_cDNA_CD8_d8_2,Gene Expression
CD8_d8_3,22NK2FLT3,3_cDNA_CD8_d8_3,Gene Expression
CD8_d8_1,22NK2FLT3,4_HTO_CD8_d8_1,Multiplexing Capture
CD8_d8_2,22NK2FLT3,5_HTO_CD8_d8_2,Multiplexing Capture
CD8_d8_3,22NK2FLT3,6_HTO_CD8_d8_3,Multiplexing Capture
</code></pre>
<h3 id="prepare-cmocsv">Prepare <code>cmo.csv</code></h3>
<pre><code>id,name,sequence,feature_type,read,pattern
A0301,A0301,ACCCACCAGTAAGAC,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0302,A0302,GGTCGAGAGCATTCA,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0303,A0303,CTTGCCGCATGTCAT,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0304,A0304,AAAGCATTCTTCACG,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0305,A0305,CTTTGTCTTTGTGAG,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0306,A0306,TATGCTGCCACGGTA,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0307,A0307,GAGTCTGCCAGTATC,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0308,A0308,TATAGAACGCCAGGC,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0309,A0309,TGCCTATGAAACAAG,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
A0310,A0310,CCGATTGTAACAGAC,Multiplexing Capture,R2,5PNNNNNNNNNN(BC)
</code></pre>
<h2 id="step-2-convert-per-sample-bam-files-to-fastqs-for-the-gex-data">Step 2: Convert per sample bam files to FASTQs for the GEX data</h2>
<p>In this step, you will run the snakemake pipeline to:
1. Convert per-sample BAM file to FASTQ files
2. Prepare the folder for next step</p>
<pre><code>conda activate /mnt/ccrsf-ifx/Software/tools/conda_env4scwf
module load samtools
snakemake -s multi_5p_bam2fastq.py --configfile config.yaml --profile ./slurm/  -e cluster-generic -j 20 
</code></pre>
<pre><code>indir: &quot;/mnt/ccrsf-static/singlecell_projects/PamelaSchwartzberg_CS037516_9sclibs_08072024/Analysis_demultiplex/&quot;
outdir: &quot;outdir&quot;
</code></pre>
<pre><code>Job stats:
job                   count
------------------  -------
all                       1
bamtofastq               30
prep_raw_fq_folder       30
total                    61
</code></pre>
<h2 id="step3-final-run-cellranger-multi-for-gex-fb-tcr-and-bcr-data">Step3: Final: Run cellranger multi for GEX, FB, TCR, and BCR data</h2>
<p>Only GEX and VDJ data is used in the example project. </p>
<ul>
<li>
<p>Prepare fastq files so that the folder structure meets what <code>cellranger mkfastq</code> outputs. </p>
</li>
<li>
<p>Prepare <code>libraries.csv</code></p>
</li>
</ul>
<p>In the example we used, there are 3 GEX libraries and each GEX library has 10 samples multiplexed together. So in total, there are 30 samples. There are 3 corresponding VDJ libraries. Since we don't perform demultiplexing using HOT on VDJ data. So for the 10 samples from a particular GEX library, same VDJ data were provided. </p>
<p>In the <code>libraries.csv</code> file below, only samples from <code>D8_d8_1</code> are shown. </p>
<pre><code>Name,Flowcell,Sample,Type,force-cells
CD8_d8_1_A0301,22NK2FLT3,CD8_d8_1_A0301,Gene Expression,1049
CD8_d8_1_A0302,22NK2FLT3,CD8_d8_1_A0302,Gene Expression,857
CD8_d8_1_A0303,22NK2FLT3,CD8_d8_1_A0303,Gene Expression,1042
CD8_d8_1_A0304,22NK2FLT3,CD8_d8_1_A0304,Gene Expression,957
CD8_d8_1_A0305,22NK2FLT3,CD8_d8_1_A0305,Gene Expression,986
CD8_d8_1_A0306,22NK2FLT3,CD8_d8_1_A0306,Gene Expression,824
CD8_d8_1_A0307,22NK2FLT3,CD8_d8_1_A0307,Gene Expression,1062
CD8_d8_1_A0308,22NK2FLT3,CD8_d8_1_A0308,Gene Expression,897
CD8_d8_1_A0309,22NK2FLT3,CD8_d8_1_A0309,Gene Expression,825
CD8_d8_1_A0310,22NK2FLT3,CD8_d8_1_A0310,Gene Expression,1179
CD8_d8_1_A0301,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,1049
CD8_d8_1_A0302,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,857
CD8_d8_1_A0303,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,1042
CD8_d8_1_A0304,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,957
CD8_d8_1_A0305,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,986
CD8_d8_1_A0306,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,824
CD8_d8_1_A0307,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,1062
CD8_d8_1_A0308,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,897
CD8_d8_1_A0309,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,825
CD8_d8_1_A0310,22NK2FLT3,7_VDJ_CD8_d8_1,VDJ-T,1179
</code></pre>
<h2 id="reference">Reference</h2>
<p><a href="https://www.10xgenomics.com/analysis-guides/demultiplexing-and-analyzing-5%E2%80%99-immune-profiling-libraries-pooled-with-hashtags">Demultiplexing and Analyzing 5’ Immune Profiling Libraries Pooled with Hashtags</a></p>
              
            </div>
          </div><footer>

  <hr/>

  <div role="contentinfo">
    <!-- Copyright etc -->
      <p>Copyright &copy; 2024 CCRSFIFX</p>
  </div>

  Built with <a href="https://www.mkdocs.org/">MkDocs</a> using a <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>.
</footer>
          
        </div>
      </div>

    </section>

  </div>

  <div class="rst-versions" role="note" aria-label="Versions">
  <span class="rst-current-version" data-toggle="rst-current-version">
    
        <span>
          <a href="https://github.com/CCRSF-IFX/scMaestro.git" class="fa fa-code-fork" style="color: #fcfcfc"> CCRSF-IFX/scMaestro</a>
        </span>
    
    
    
  </span>
</div>
    <script src="../js/jquery-3.6.0.min.js"></script>
    <script>var base_url = "..";</script>
    <script src="../js/theme_extra.js"></script>
    <script src="../js/theme.js"></script>
      <script src="../search/main.js"></script>
    <script>
        jQuery(function () {
            SphinxRtdTheme.Navigation.enable(true);
        });
    </script>

</body>
</html>
